pkg/wallet/wallet.go

```diff
@@ -123,8 +130,44 @@ func (w *BaseWallet) Close() error {
 }
 
 // CreateDID returns new DID Document
-func (w *BaseWallet) CreateDID() (*did.Doc, error) {
-	return nil, fmt.Errorf("To be implemented")
+func (w *BaseWallet) CreateDID(method string) (*did.Doc, error) {
+	// Generate key pair
+	pub, err := w.CreateKey()
+	if err != nil {
+		return nil, fmt.Errorf("failed to create DID: %w", err)
+	}
+
+	// DID identifier
+	id := fmt.Sprintf(didFormat, method, pub[:16])
+
+	// Supporting only one public key now
+	pubKey := did.PublicKey{
+		ID: fmt.Sprintf(didPKID, id, 1),
+		//TODO hardcoding public key type for now
+		// Should be dynamic for multi-key support
+		Type:       "Ed25519VerificationKey2018",
+		Controller: id,
+		Value:      []byte(pub),
+	}
+
+	//Service endpoints
+	service := did.Service{
+		ID:              fmt.Sprintf(didServiceID, id, 1),
+		Type:            "InboundTransport",
```

Comment:
`Service.Type` should be `did-communication` as defined [here](https://github.com/hyperledger/aries-rfcs/tree/master/features/0067-didcomm-diddoc-conventions#did-documents)

Reply:
As discussed, removed hardcoded service type value, instead it can be passed by caller as an option.